src/Infrastructure/Component/PreAppliedFilters.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fau\DegreeProgram\Output\Infrastructure\Component;
+
+use Fau\DegreeProgram\Common\Infrastructure\TemplateRenderer\Renderer;
+
+/**
+ * @psalm-type PreAppliedFilters = array{
+ *     preAppliedFilters: array<string, list<string|int>>,
+ * }
+ */
+class PreAppliedFilters implements RenderableComponent
+{
+    private const DEFAULT_ATTRIBUTES = [
+        'preAppliedFilters' => [],
+    ];
+
+    public function __construct(
+        private Renderer $renderer,
+    ) {
+    }
+
+    public function render(array $attributes = self::DEFAULT_ATTRIBUTES): string
+    {
+        /** @var PreAppliedFilters $attributes */
+        $attributes = wp_parse_args($attributes, self::DEFAULT_ATTRIBUTES);
+        $preAppliedFilters = [];
+
+        foreach ($attributes['preAppliedFilters'] as $name => $values) {
+            foreach ($values as $value) {
```

Comment:
Sorry if this is obvious, but how do we ensure that `$values` is iteratable? Can't it be a simple string or numeric value?

Reply:
Thanks for pointing this out, the `$values` variable is always an array:

https://github.com/RRZE-Webteam/FAU-Studium-Embed/blob/74dac24a6c7a0c2e26cbfaa6daccba94f5561448/src/Infrastructure/Shortcode/ShortcodeAttributesNormalizer.php#L92-L100

https://github.com/RRZE-Webteam/FAU-Studium-Embed/blob/74dac24a6c7a0c2e26cbfaa6daccba94f5561448/src/Infrastructure/Shortcode/ShortcodeAttributesNormalizer.php#L115